tasks/main.yml

```diff
@@ -50,6 +50,18 @@
     - network_provider == "nm"
   no_log: true
 
+# If user switch from initscripts provider where NM_CONTROLLED=no defined in
+# ifcfg-ethX file, NetworkManager daemon will treat that interface as
+# strickly unmanaged, even the follow up deletion of ifcfg-ethX file cannot
+# change the NetworManager's unmanaged state of this interface. This will
+# prevent any follow up "nm" provider action on this interface.
+# To solve that, we instruct NetworkManager to reload the configuration.
+- name: Reload NetworkManager configuration
+  command: nmcli connection reload
```

Comment:
What is the reason to do this with nmcli instead of doing it via libnm in the python? Also, this specific command might have side effects beyond the interfaces that are specified in network_interfaces, so it might not be a good idea. I believe it has been considered in the past. @thom311 what's your opinion?

Reply:
Changed to use `libnm` to reload the configuration when on NM 1.18.